src/python/pants/backend/makeself/makeself.py

```diff
@@ -0,0 +1,322 @@
+
+logger = logging.getLogger(__name__)
+
+
+class MakeselfSubsystem(TemplatedExternalTool):
+    options_scope = "makeself"
+    help = "A tool to generate a self-extractable compressed tar archives."
+
+    default_version = "2.5.0"
+    default_known_versions = [
+        "2.5.0|macos_arm64 |4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+        "2.5.0|macos_x86_64|4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+        "2.5.0|linux_arm64 |4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+        "2.5.0|linux_x86_64|4d2fa9d898be22c63bb3c6bb7cc3dc97237700dea6d6ad898dcbec0289df0bc4|45867",
+    ]
+
+    default_url_template = "https://github.com/megastep/makeself/releases/download/release-{version}/makeself-{version}.run"
+
+
+@dataclass(frozen=True)
+class RunMakeselfArchive:
+    exe: str
+    input_digest: Digest
+    description: str
+    level: LogLevel = LogLevel.INFO
+    output_directory: Optional[str] = None
+
+
+@rule(desc="Run makeself archive", level=LogLevel.DEBUG)
+async def run_makeself_archive(
+    request: RunMakeselfArchive,
+    awk: AwkBinary,
+    base64: Base64Binary,
+    basename: BasenameBinary,
+    bash: BashBinary,
+    bzip2: Bzip2Binary,
+    cat: CatBinary,
+    cut: CutBinary,
+    dd: DdBinary,
+    df: DfBinary,
+    dirname: DirnameBinary,
+    expr: ExprBinary,
+    find: FindBinary,
+    gpg: GpgBinary,
+    gzip: GzipBinary,
+    head: HeadBinary,
+    id: IdBinary,
+    md5sum: Md5sumBinary,
+    mkdir: MkdirBinary,
+    pwd: PwdBinary,
+    rm: RmBinary,
+    sed: SedBinary,
+    shasum: DateBinary,
+    tail: TailBinary,
+    tar: TarBinary,
+    test: TestBinary,
+    wc: WcBinary,
+    xz: XzBinary,
+    zstd: ZstdBinary,
+) -> Process:
+    shims = await Get(
+        BinaryShims,
+        BinaryShimsRequest(
+            paths=(
+                awk,
+                base64,
+                basename,
+                bash,
+                bzip2,
+                cat,
+                cut,
+                dd,
+                df,
+                dirname,
+                expr,
+                find,
+                gpg,
+                gzip,
+                head,
+                id,
+                md5sum,
+                mkdir,
+                pwd,
+                rm,
+                sed,
+                shasum,
+                tail,
+                tar,
+                test,
+                wc,
+                xz,
+                zstd,
+            ),
+            rationale="run makeself archive",
+        ),
+    )
+    output_directories = []
+    argv = [
+        request.exe,
+        "--accept",
+        "--noprogress",
+        "--nox11",
+        "--nochown",
+        "--nodiskspace",
+        "--quiet",
+    ]
+
+    if output_directory := request.output_directory:
+        output_directories = [output_directory]
+        argv.extend(["--keep", "--target", request.output_directory])
```

Comment:
Is this to allow a performance optimisation of only extracting once? Should this also pass `--noexec`?

Reply:
Right. So, to confirm, you're saying all callers of this who pass `output_directory` definitely want the start-up script to be run?